### PR TITLE
Feature/storage

### PR DESCRIPTION
**Pull request by arch89**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -25,6 +25,9 @@ lazy val core = (project in file("core"))
       "org.scala-lang" % "scala-reflect" % scalaVersion.value,
       "org.scala-lang" % "scala-compiler" % scalaVersion.value
     ),
+    libraryDependencies += "com.beachape" %% "enumeratum" % Version.enumeratum,
+    libraryDependencies += "com.github.mauricio" %% "postgresql-async" % "0.2.21",
```
Comment (c0d3r85) [inline build.sbt]: Version.postgresqlAsync

File: core/src/main/scala/ru/tinkoff/codefest/telegram/BotCommand.scala [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+sealed trait BotCommand extends EnumEntry with Lowercase
+
+object BotCommand extends Enum[BotCommand] {
+
+  case object Reset extends BotCommand
+  case object State extends BotCommand
+
+  val values = findValues
+
+  private val pattern = s"/(${values.map(_.entryName).mkString("|")})".r
```
Comment (c0d3r85) [inline core/src/main/scala/ru/tinkoff/codefest/telegram/BotCommand.scala:16]: lets's ignore leading spaces

File: core/src/main/scala/ru/tinkoff/codefest/executor/Interpretator.scala [hunk 3]
```diff
@@ -1,41 +1,61 @@
 package ru.tinkoff.codefest.executor
 
 import java.io.{ByteArrayOutputStream, FileInputStream, PrintWriter}
+import java.lang.System.{lineSeparator => EOL}
```
Comment (c0d3r85) [inline core/src/main/scala/ru/tinkoff/codefest/executor/Interpretator.scala]: not sure about this
EOL now is server-dependent, but message comes from client

File: core/src/main/scala/ru/tinkoff/codefest/storage/Storage.scala [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package ru.tinkoff.codefest.storage
+
+import scala.language.higherKinds
```
Comment (c0d3r85) [inline core/src/main/scala/ru/tinkoff/codefest/storage/Storage.scala]: scala compiler options already has it

**Comment by c0d3r85** [comment]
There is some Russian commit messages, that breaks commit history

File: core/src/main/scala/ru/tinkoff/codefest/storage/Storage.scala (hunk 4 above)
Comment (danslapman) [reply core/src/main/scala/ru/tinkoff/codefest/storage/Storage.scala]: I think this line is to please IDEA

File: akka-http/src/main/scala/ru/tinkoff/codefest/http/Main.scala [hunk 5]
```diff
@@ -17,13 +17,18 @@ object Main extends IOApp {
       FunctionK.lift[IO, Future](unsafeToFuture)
     }
 
+    implicit val nt2: Future ~> IO = {
```
Comment (danslapman) [inline akka-http/src/main/scala/ru/tinkoff/codefest/http/Main.scala:20]: How about something like this (using kind-projector)
`λ[Future ~> IO](IO.fromFuture)`
?

File: core/src/main/scala/ru/tinkoff/codefest/storage/postgresql/PostgreSQLStorage.scala [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+import cats.~>
+import scala.language.higherKinds
+import ru.tinkoff.codefest.storage.Storage
+import ru.tinkoff.codefest.storage.postgresql.queries._
+
+class PostgreSQLStorage[F[_]](implicit ec: ExecutionContext, nt: Future ~> F) extends Storage[F] {
+  import ctx._
+  override def save(chatId: Long, state: Option[String]): F[Unit] =
+    nt(ctx.run(upsertChatQuery(Chat(chatId, state))).map(_ => ()))
```
Comment (danslapman) [inline core/src/main/scala/ru/tinkoff/codefest/storage/postgresql/PostgreSQLStorage.scala:13]: https://github.com/typelevel/mouse/blob/master/shared/src/main/scala/mouse/anyf.scala#L9 ?

File: core/src/main/scala/ru/tinkoff/codefest/storage/Storage.scala (hunk 4 above)
Comment (arch89) [reply core/src/main/scala/ru/tinkoff/codefest/storage/Storage.scala]: seems that explicit flag had already pleased IDEA

File: akka-http/src/main/scala/ru/tinkoff/codefest/http/Main.scala (hunk 5 above)
Comment (arch89) [reply akka-http/src/main/scala/ru/tinkoff/codefest/http/Main.scala:20]: ok

File: core/src/main/scala/ru/tinkoff/codefest/telegram/BotCommand.scala (hunk 2 above)
Comment (arch89) [reply core/src/main/scala/ru/tinkoff/codefest/telegram/BotCommand.scala:16]: done

File: build.sbt (hunk 1 above)
Comment (arch89) [reply build.sbt]: done

**Comment by c0d3r85** [comment]
LGTM

**Pull request merged**